network/base/views.py

```diff
@@ -384,17 +382,31 @@ def prediction_windows(request, sat_id, start_date, end_date, station_id=None):
 def observation_view(request, id):
     """View for single observation page."""
     observation = get_object_or_404(Observation, id=id)
-    data = Data.objects.filter(observation=observation)
+    dataset = Data.objects.filter(observation=observation)
 
     # not all users will be able to vet data within an observation, allow
     # staff, observation requestors, and station owners
     is_vetting_user = False
     if request.user.is_authenticated():
         if request.user == observation.author or \
-            data.filter(ground_station__in=Station.objects.filter(owner=request.user)).count or \
+            dataset.filter(
+                ground_station__in=Station.objects.filter(owner=request.user)).count or \
                 request.user.is_staff:
                     is_vetting_user = True
 
+    # Determine if there is no valid payload file in the observation dataset
+    if request.user.is_staff:
+        data_payload_exists = False
+        for data in dataset:
+            if data.payload_exists:
+                data_payload_exists = True
+    # This context flag will determine if a delete button appears for the observation.
+    is_deletable = False
+    if observation.author == request.user and observation.is_deletable:
+        is_deletable = True
+    if request.user.is_staff and not data_payload_exists:
```

Comment:
Wouldn't that give delete permissions to all staff, regardless if the have admin rights or not?